### PR TITLE
Update FormattedHeader to support compact layout on mobile

### PR DESCRIPTION
**Pull request by delawski**
So far the `<FormattedHeader />` displayed in almost the same way no matter what the screen size is. This PR adds a `compactOnMobile ` flag to the component so that it can be rendered in a compact way on smaller screen sizes.

We need that in order to implement new Jetpack single products.

**Larger screen sizes:**

Screenshot 2019-10-23 at 14 39 30

**Small screens:**

Screenshot 2019-10-23 at 14 39 37

#### Changes proposed in this Pull Request

* Add `compactOnMobile ` flag to the `<FormattedHeader />`
* Update docs and example file
* Use new responsive header in the `<PlansFeaturesMain />` component

#### Testing instructions

* Check out this branch
* Go to http://calypso.localhost:3000/plans/ and select one of you Jetpack sites
* Observe if the formatted header ("Single Products") changes its layout and font size based on the screen size

Fixes n/a

### CODE REVIEW

**Comment by enejb** [comment]
Instead of responsive. can we call it `compact` or something else instead. 
I think that would make more sense. 

I though it mean something else. 🤔 

Marin do you have any good naming suggestions here? 
Do we have any other patterns in the current implementation of components that implements something like this?

**Comment by enejb** [comment]
This works well. In my limited testing!

**Comment by delawski** [comment]
@enejb Thank you for your review!

I was hesitant to call it just `compact` (as I was considering that name, too). It's not really *compact* since on desktop it looks exactly the same as the original `<FormattedHeader />`. The difference is that for smaller screens it becomes compact. That's where the `responsive` prop name came from.

@tyxla Any suggestions for the name?

File: client/components/formatted-header/README.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ FormattedHeader (JSX)
 
 This component displays a header and optional sub-header.
 
+When the `responsive` flag is set, the header renders in a compact way on smaller screen sizes.
```
Comment (tyxla) [inline client/components/formatted-header/README.md]: I have second thoughts about the name of this flag. Considering that it makes the header "compact" does it make sense to rename it to `compact`? "Responsive" seems to be too broad in my head.

File: client/components/formatted-header/style.scss [hunk 2]
```diff
@@ -36,3 +36,20 @@
 		padding: 0 20px 24px;
 	}
 }
+
+// Responsive header
+.formatted-header.is-responsive {
```
Comment (tyxla) [inline client/components/formatted-header/style.scss]: Nit: does it make sense the following code style instead? No hard preference, just providing an alternative.

```
.formatted-header {
	&.is-responsive {
		...
	}
}
```

**Comment by tyxla** [comment]
> @tyxla Any suggestions for the name?

Maybe `compactOnMobile`?

File: client/components/formatted-header/style.scss (hunk 2 above)
Comment (delawski) [reply client/components/formatted-header/style.scss]: I didn't want to create another nesting level here. There's already a media query and child selectors inside so we're getting pretty deep in terms of nesting already (3 levels while the recommendation it to have 2 levels at most).

Comment (tyxla) [reply client/components/formatted-header/style.scss]: That's fair, let's keep it as-is then 👍

File: client/components/formatted-header/docs/example.jsx [hunk 3]
```diff
@@ -16,10 +16,17 @@ export default class FormattedHeaderExample extends PureComponent {
...
-				headerText="This is the header."
-				subHeaderText="This is the optional subheader."
-			/>
+			<Fragment>
+				<FormattedHeader
+					headerText="This is the header."
+					subHeaderText="This is the optional subheader."
+				/>
+				<FormattedHeader
+					headerText="This is the responsive header."
```
Comment (tyxla) [inline client/components/formatted-header/docs/example.jsx]:
```suggestion
					headerText="This is the compact on mobile header."
```

**Pull request merged**